packages/fiori/src/NavigationLayout.ts

```diff
@@ -0,0 +1,120 @@
+import UI5Element from "@ui5/webcomponents-base/dist/UI5Element.js";
+import customElement from "@ui5/webcomponents-base/dist/decorators/customElement.js";
+import property from "@ui5/webcomponents-base/dist/decorators/property.js";
+import slot from "@ui5/webcomponents-base/dist/decorators/slot.js";
+import litRender from "@ui5/webcomponents-base/dist/renderer/LitRenderer.js";
+import browserScrollbarCSS from "@ui5/webcomponents/dist/generated/themes/BrowserScrollbar.css.js";
+import {
+	isPhone,
+	isTablet,
+	isCombi,
+} from "@ui5/webcomponents-base/dist/Device.js";
+import type SideNavigation from "./SideNavigation.js";
+
+// Template
+import NavigationLayoutTemplate from "./generated/templates/NavigationLayoutTemplate.lit.js";
+
+// Styles
+import NavigationLayoutCss from "./generated/themes/NavigationLayout.css.js";
+
+/**
+ * @class
+ *
+ * ### Overview
+ *
+ * The `ui5-navigation-layout` is a container, used to create a navigation
+ * layout that includes a header, side navigation, and content area.
+ * ### ES6 Module Import
+ *
+ * `import "@ui5/webcomponents-fiori/dist/NavigationLayout.js";`
+ * @constructor
+ * @extends UI5Element
+ * @since 2.4.0
+ * @public
+ */
+@customElement({
+	tag: "ui5-navigation-layout",
+	languageAware: true,
+	renderer: litRender,
+	styles: [
+		browserScrollbarCSS,
+		NavigationLayoutCss,
+	],
+	template: NavigationLayoutTemplate,
+})
+class NavigationLayout extends UI5Element {
+	_sideCollapsed = isPhone() || (isTablet() && !isCombi());
+
+	/**
+	 * @private
+	 */
+	@property({ type: Boolean })
+	isPhone = isPhone();
+
+	/**
+	 * @private
+	 */
+	@property({ type: Boolean })
+	isTablet = isTablet() && !isCombi();
+
+	/**
+	 * Indicates whether the side menu is collapsed.
+	 * @default false
+	 * @public
+	 */
+	@property({ type: Boolean })
+	set sideCollapsed(value: boolean) {
+		this._sideCollapsed = value;
+
+		if (isPhone()) {
+			return;
+		}
+
+		const sideNavigation = this.sideContent[0] as SideNavigation;
+
+		if (sideNavigation) {
```

Comment:
better to check with instanceof ?
e.g.
const sideContent = this.sideContent[0];

if (sideContent instanceof SideNavigation) {
	sideContent.collapsed = value;
}

this will avoid problems if the sideContent is not SideNavigation

Reply:
After consulting - it seems that "instanceof" is not good to be used in webcomponents.
Better use a check for the "collapsed" property 
const sideContent = this.sideContent[0] as SideNavigation;
if ("collapsed" in sideContent) {
	sideContent.collapsed = this.sideCollapsed;
}
		
or more strict - do a method as in UI5Element.ts line 1348 - combined with new method isSideNavigation.

Reply:
an alternative is also to limit the sideContent to allow only SideNavigation